ASTImporter, unimplemented DependentTemplateSpecialization Type visitation

```
ArpInterface.cc_d8f953b6e736cd3e4069dfea22a426b1.plist.zip
Archive:  ArpInterface.cc_d8f953b6e736cd3e4069dfea22a426b1.plist.zip
 extracting: stderr                  
ANALYZE (CTU loaded AST for source file): /local/scratch/repos/ML66/reports/bns__2017_11_16__15_34_00/ctu-dir/i386/ast/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/src/platform/eqm/operation/src/PSInterface.cc
warning: cannot import unsupported AST node DependentTemplateSpecialization
warning: cannot import unsupported AST node DependentTemplateSpecialization
warning: cannot import unsupported AST node DependentTemplateSpecialization
warning: cannot import unsupported AST node DependentTemplateSpecialization
warning: cannot import unsupported AST node DependentTemplateSpecialization
warning: cannot import unsupported AST node DependentTemplateSpecialization
warning: cannot import unsupported AST node DependentTemplateSpecialization
warning: cannot import unsupported AST node DependentTemplateSpecialization
warning: cannot import unsupported AST node DependentTemplateSpecialization
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(llvm::sys::PrintStackTrace(llvm::raw_ostream&)+0x1a)[0x1c8e8da]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(llvm::sys::RunSignalHandlers()+0x3e)[0x1c8c85e]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0[0x1c8c982]
/lib64/libpthread.so.0[0x3b7bc0f7e0]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::CXXUnresolvedConstructExpr::CXXUnresolvedConstructExpr(clang::TypeSourceInfo*, clang::SourceLocation, llvm::ArrayRef<clang::Expr*>, clang::SourceLocation)+0x20)[0x355c220]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::CXXUnresolvedConstructExpr::Create(clang::ASTContext const&, clang::TypeSourceInfo*, clang::SourceLocation, llvm::ArrayRef<clang::Expr*>, clang::SourceLocation)+0x108)[0x355c508]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitCXXUnresolvedConstructExpr(clang::CXXUnresolvedConstructExpr*)+0x273)[0x3491913]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Stmt*)+0xb5)[0x347c4b5]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitParenListExpr(clang::ParenListExpr*)+0x9f)[0x349265f]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Stmt*)+0xb5)[0x347c4b5]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::CXXCtorInitializer*)+0x1d)[0x349484d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitFunctionDecl(clang::FunctionDecl*)+0xb9d)[0x347a2fd]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*)+0x8f)[0x348195f]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDefinition(clang::RecordDecl*, clang::RecordDecl*, clang::ASTNodeImporter::ImportDefinitionKind)+0x798)[0x3480ec8]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0x75)[0x347ab45]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*)+0x121)[0x3485041]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitRecordDecl(clang::RecordDecl*)+0x85a)[0x348731a]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*)+0x121)[0x3485041]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*)+0x8f)[0x348195f]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*)+0xd4)[0x3484624]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::ImportContext(clang::DeclContext*)+0x2c)[0x348167c]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&)+0x37)[0x34836b7]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*)+0xd5)[0x3484ff5]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::TemplateName)+0x100)[0x3485dc0]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitTemplateSpecializationType(clang::TemplateSpecializationType const*)+0x1a)[0x348640a]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0[0x347da85]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::QualType)+0x12f)[0x347dcef]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitTypedefNameDecl(clang::TypedefNameDecl*, bool)+0x245)[0x34896d5]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*)+0x8f)[0x348195f]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDefinition(clang::RecordDecl*, clang::RecordDecl*, clang::ASTNodeImporter::ImportDefinitionKind)+0x798)[0x3480ec8]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0x75)[0x347ab45]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*)+0x121)[0x3485041]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitRecordDecl(clang::RecordDecl*)+0x85a)[0x348731a]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*)+0x121)[0x3485041]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*)+0x8f)[0x348195f]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*)+0xd4)[0x3484624]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::ImportContext(clang::DeclContext*)+0x2c)[0x348167c]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&)+0x37)[0x34836b7]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*)+0xd5)[0x3484ff5]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0[0x348b19b]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::ImportContext(clang::DeclContext*)+0x2c)[0x348167c]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&)+0x37)[0x34836b7]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitTypedefNameDecl(clang::TypedefNameDecl*, bool)+0x4e)[0x34894de]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitTypedefType(clang::TypedefType const*)+0x10)[0x347b3b0]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0[0x347db65]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::QualType)+0x12f)[0x347dcef]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitCallExpr(clang::CallExpr*)+0x20)[0x3491bc0]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Stmt*)+0xb5)[0x347c4b5]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitExprWithCleanups(clang::ExprWithCleanups*)+0x20)[0x3491050]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Stmt*)+0xb5)[0x347c4b5]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDefinition(clang::VarDecl*, clang::VarDecl*, clang::ASTNodeImporter::ImportDefinitionKind)+0x4f)[0x347d69f]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitVarDecl(clang::VarDecl*)+0x51c)[0x348d81c]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::ImportDeclGroup(clang::DeclGroupRef)+0xc5)[0x347b765]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitDeclStmt(clang::DeclStmt*)+0x19)[0x347b879]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Stmt*)+0xb5)[0x347c4b5]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitCompoundStmt(clang::CompoundStmt*)+0x9e)[0x347cb3e]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Stmt*)+0xb5)[0x347c4b5]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTNodeImporter::VisitFunctionDecl(clang::FunctionDecl*)+0x7b1)[0x3479f11]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ASTImporter::Import(clang::Decl*)+0xcd)[0x347ab9d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::tooling::CrossTranslationUnit::getCrossTUDefinition(clang::FunctionDecl const*, llvm::StringRef, llvm::StringRef, llvm::StringRef, bool)+0x120)[0x2c3b000]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::AnyFunctionCall::getRuntimeDefinition() const+0xfb)[0x2b71dfb]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::CXXInstanceCall::getRuntimeDefinition() const+0x8b)[0x2b72d7b]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::CXXMemberCall::getRuntimeDefinition() const+0x3a)[0x2b730da]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::ExprEngine::defaultEvalCall(clang::ento::NodeBuilder&, clang::ento::ExplodedNode*, clang::ento::CallEvent const&)+0x37a)[0x2bd561a]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::CheckerManager::runCheckersForEvalCall(clang::ento::ExplodedNodeSet&, clang::ento::ExplodedNodeSet const&, clang::ento::CallEvent const&, clang::ento::ExprEngine&)+0x3be)[0x2b7e7de]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::ExprEngine::evalCall(clang::ento::ExplodedNodeSet&, clang::ento::ExplodedNode*, clang::ento::CallEvent const&)+0xf0)[0x2bc8f30]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::ExprEngine::VisitCallExpr(clang::CallExpr const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&)+0x1c0)[0x2bc9150]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::ExprEngine::Visit(clang::Stmt const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&)+0x401)[0x2bb7bc1]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::ExprEngine::ProcessStmt(clang::CFGStmt, clang::ento::ExplodedNode*)+0x1fc)[0x2bb98cc]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::ExprEngine::processCFGElement(clang::CFGElement, clang::ento::ExplodedNode*, unsigned int, clang::ento::NodeBuilderContext*)+0xfc)[0x2bb9afc]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::CoreEngine::HandlePostStmt(clang::CFGBlock const*, unsigned int, clang::ento::ExplodedNode*)+0x5a)[0x2b878ba]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::CoreEngine::dispatchWorkItem(clang::ento::ExplodedNode*, clang::ProgramPoint, clang::ento::WorkListUnit const&)+0x18c)[0x2b87a5c]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ento::CoreEngine::ExecuteWorkList(clang::LocationContext const*, unsigned int, llvm::IntrusiveRefCntPtr<clang::ento::ProgramState const>)+0x19b)[0x2b8838b]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0[0x28b7132]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0[0x28b7c01]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0[0x28c892f]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::MultiplexConsumer::HandleTranslationUnit(clang::ASTContext&)+0x28)[0x21f9bc8]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ParseAST(clang::Sema&, bool, bool)+0x2e8)[0x2ca1108]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::FrontendAction::Execute()+0xf6)[0x21cafe6]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::CompilerInstance::ExecuteAction(clang::FrontendAction&)+0x136)[0x219a7a6]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(clang::ExecuteCompilerInvocation(clang::CompilerInstance*)+0xd8b)[0x2265dab]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(cc1_main(llvm::ArrayRef<char const*>, char const*, void*)+0x9f8)[0x82a4d8]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0(main+0x18bb)[0x827efb]
/lib64/libc.so.6(__libc_start_main+0xfd)[0x3b7b81ed1d]
/proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0[0x825039]
Stack dump:
0.	Program arguments: /proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin/clang-5.0 -cc1 -triple powerpc-montavista-linux-gnu -analyze -disable-free -disable-llvm-verifier -discard-value-names -main-file-name ArpInterface.cc -analyzer-store=region -analyzer-opt-analyze-nested-blocks -analyzer-eagerly-assume -analyzer-output plist -w -mrelocation-model static -mthread-model posix -fmath-errno -masm-verbose -mconstructor-aliases -target-cpu ppc -mfloat-abi hard -momit-leaf-frame-pointer -dwarf-column-info -debugger-tuning=gdb -resource-dir /proj/tn/tools/code_checker/CodeChecker_6.2.0/include -isystem /proj/tn/tools/code_checker/CodeChecker_6.2.0/include -isystem /proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/libexec/../lib/gcc/powerpc64-montavista-linux-gnu/4.7.0/../../../../powerpc64-montavista-linux-gnu/include/c++/4.7.0 -isystem /proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/libexec/../lib/gcc/powerpc64-montavista-linux-gnu/4.7.0/../../../../powerpc64-montavista-linux-gnu/include/c++/4.7.0/powerpc64-montavista-linux-gnu/32 -isystem /proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/libexec/../lib/gcc/powerpc64-montavista-linux-gnu/4.7.0/../../../../powerpc64-montavista-linux-gnu/include/c++/4.7.0/backward -isystem /proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/libexec/../lib/gcc/powerpc64-montavista-linux-gnu/4.7.0/include -isystem /proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/libexec/../lib/gcc/powerpc64-montavista-linux-gnu/4.7.0/include-fixed -isystem /proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/libexec/../lib/gcc/powerpc64-montavista-linux-gnu/4.7.0/../../../../powerpc64-montavista-linux-gnu/include -isystem /proj/tn/build/sysroot/e5500/R1A04/usr/include -D ARCH_e5500 -D BNS_APP_SECURITY_FEATURE -D PRODTYPE_NPU -I /local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/include -I /local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/include/lch -I /local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/include/rph -I /local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/include/ecps -I /local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/include/tacacs -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/envoy/target/h -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/xf -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/cmo -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/smo -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/tm -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/tnu -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/radius -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/ecps -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/eqm -I /local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/platform/lch/lh/include -I -I -I / -I -I -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include -I /local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/lch -D RELEASE -D _REENTRANT -D USE_SNMP -isysroot /proj/tn/build/sysroot/e5500/R1A04 -internal-isystem /proj/tn/build/sysroot/e5500/R1A04/usr/local/include -internal-isystem /proj/tn/tools/code_checker/CodeChecker_6.2.0/include/include -internal-externc-isystem /proj/tn/build/sysroot/e5500/R1A04/include -internal-externc-isystem /proj/tn/build/sysroot/e5500/R1A04/usr/include -O2 -fdeprecated-macro -fdebug-compilation-dir /local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/make -ferror-limit 19 -fmessage-length 0 -fno-signed-char -fobjc-runtime=gcc -fcxx-exceptions -fexceptions -fdiagnostics-show-option -vectorize-loops -vectorize-slp -plugin checkercfg -load /proj/tn/tools/code_checker/CodeChecker_6.2.0/plugin/libericsson-checkers-5.2-5.0.0.so -analyzer-opt-analyze-headers -analyzer-output=plist-multi-file -analyzer-disable-checker alpha.clone.CloneChecker -analyzer-disable-checker alpha.core.BoolAssignment -analyzer-disable-checker alpha.core.CallAndMessageUnInitRefArg -analyzer-disable-checker alpha.core.CastSize -analyzer-disable-checker alpha.core.CastToStruct -analyzer-disable-checker alpha.core.Conversion -analyzer-disable-checker alpha.core.DynamicTypeChecker -analyzer-disable-checker alpha.core.FixedAddr -analyzer-disable-checker alpha.core.IdenticalExpr -analyzer-disable-checker alpha.core.PointerArithm -analyzer-disable-checker alpha.core.PointerSub -analyzer-disable-checker alpha.core.SizeofPtr -analyzer-disable-checker alpha.core.TestAfterDivZero -analyzer-disable-checker alpha.cplusplus.IteratorRange -analyzer-disable-checker alpha.cplusplus.MisusedMovedObject -analyzer-disable-checker alpha.deadcode.UnreachableCode -analyzer-disable-checker alpha.ericsson.cpp.BitwiseShift -analyzer-disable-checker alpha.ericsson.cpp.CtorUninitializedMember -analyzer-disable-checker alpha.ericsson.cpp.DirtyScalar -analyzer-disable-checker alpha.ericsson.cpp.ExceptionMisuse -analyzer-disable-checker alpha.ericsson.cpp.InvariablePtrBranch -analyzer-disable-checker alpha.ericsson.cpp.IteratorOutOfRange -analyzer-disable-checker alpha.ericsson.cpp.MismatchedIterator -analyzer-disable-checker alpha.ericsson.cpp.MissingHeaderGuard -analyzer-disable-checker alpha.ericsson.cpp.OStreamFormat -analyzer-disable-checker alpha.ericsson.cpp.PointerArithmetic -analyzer-disable-checker alpha.ericsson.cpp.Slicing -analyzer-disable-checker alpha.ericsson.cpp.stl.MissingIteratorAdapter -analyzer-disable-checker alpha.ericsson.cpp.style.PreferInit -analyzer-disable-checker alpha.ericsson.statisticsbased.SpecialReturnValue -analyzer-disable-checker alpha.ericsson.statisticsbased.UncheckedReturnValue -analyzer-disable-checker alpha.ericsson.statisticscollector.ReturnValueCheck -analyzer-disable-checker alpha.ericsson.statisticscollector.SpecialReturnValue -analyzer-disable-checker alpha.osx.cocoa.DirectIvarAssignment -analyzer-disable-checker alpha.osx.cocoa.DirectIvarAssignmentForAnnotatedFunctions -analyzer-disable-checker alpha.osx.cocoa.InstanceVariableInvalidation -analyzer-disable-checker alpha.osx.cocoa.MissingInvalidationMethod -analyzer-disable-checker alpha.osx.cocoa.localizability.PluralMisuseChecker -analyzer-disable-checker alpha.security.ArrayBound -analyzer-disable-checker alpha.security.ArrayBoundV2 -analyzer-disable-checker alpha.security.MallocOverflow -analyzer-disable-checker alpha.security.ReturnPtrRange -analyzer-disable-checker alpha.security.taint.TaintPropagation -analyzer-disable-checker alpha.unix.BlockInCriticalSection -analyzer-disable-checker alpha.unix.Chroot -analyzer-disable-checker alpha.unix.PthreadLock -analyzer-disable-checker alpha.unix.SimpleStream -analyzer-disable-checker alpha.unix.Stream -analyzer-disable-checker alpha.unix.cstring.BufferOverlap -analyzer-disable-checker alpha.unix.cstring.NotNullTerminated -analyzer-disable-checker alpha.unix.cstring.OutOfBounds -analyzer-checker=apiModeling.google.GTest -analyzer-checker=core.CallAndMessage -analyzer-checker=core.DivideZero -analyzer-checker=core.DynamicTypePropagation -analyzer-checker=core.NonNullParamChecker -analyzer-checker=core.NullDereference -analyzer-checker=core.StackAddressEscape -analyzer-checker=core.UndefinedBinaryOperatorResult -analyzer-checker=core.VLASize -analyzer-checker=core.builtin.BuiltinFunctions -analyzer-checker=core.builtin.NoReturnFunctions -analyzer-checker=core.uninitialized.ArraySubscript -analyzer-checker=core.uninitialized.Assign -analyzer-checker=core.uninitialized.Branch -analyzer-checker=core.uninitialized.CapturedBlockVariable -analyzer-checker=core.uninitialized.UndefReturn -analyzer-checker=cplusplus.NewDelete -analyzer-checker=cplusplus.NewDeleteLeaks -analyzer-checker=cplusplus.SelfAssignment -analyzer-checker=deadcode.DeadStores -analyzer-disable-checker debug.AnalysisOrder -analyzer-disable-checker debug.ConfigDumper -analyzer-disable-checker debug.DumpCFG -analyzer-disable-checker debug.DumpCallGraph -analyzer-disable-checker debug.DumpCalls -analyzer-disable-checker debug.DumpDominators -analyzer-disable-checker debug.DumpLiveVars -analyzer-disable-checker debug.DumpTraversal -analyzer-disable-checker debug.ExprInspection -analyzer-disable-checker debug.Stats -analyzer-disable-checker debug.TaintTest -analyzer-disable-checker debug.ViewCFG -analyzer-disable-checker debug.ViewCallGraph -analyzer-disable-checker debug.ViewExplodedGraph -analyzer-disable-checker ericsson.cpp.InvalidatedIterator -analyzer-checker=ericsson.cpp.MisuseEnumAsCondition -analyzer-checker=ericsson.cpp.PredWithState -analyzer-checker=ericsson.cpp.RaiiMisuse -analyzer-checker=ericsson.cpp.StrlenMisuseAlloc -analyzer-checker=ericsson.cpp.SuspiciousPointerArithmetic -analyzer-checker=ericsson.cpp.SuspiciousSizeof -analyzer-checker=ericsson.cpp.VirtualDtorPoly -analyzer-checker=ericsson.cpp.stl.ContainerOfAutoptr -analyzer-checker=ericsson.cpp.stl.MisusedRemove -analyzer-checker=ericsson.cpp.stl.PolymorphContainer -analyzer-checker=ericsson.cpp.stl.RevIterBase -analyzer-checker=ericsson.cpp.style.CatchByValue -analyzer-disable-checker ericsson.cpp.style.Goto -analyzer-checker=ericsson.cpp.style.LargeObjectPassed -analyzer-disable-checker ericsson.cpp.style.MagicNumbers -analyzer-disable-checker ericsson.cpp.style.MissingConst -analyzer-disable-checker ericsson.cpp.style.New -analyzer-disable-checker ericsson.cpp.style.NonPrivateData -analyzer-disable-checker ericsson.cpp.style.ReduceScope -analyzer-disable-checker ericsson.cpp.style.ReinterpretCast -analyzer-disable-checker ericsson.linuxkernelstyle.PointerDecl -analyzer-disable-checker ericsson.misrac.AssignmentInCondition -analyzer-disable-checker ericsson.misrac.ExternalArrayWithUnknownSize -analyzer-disable-checker ericsson.misrac.FunctionsWithNoParam -analyzer-disable-checker ericsson.misrac.IfFollowedByCompound -analyzer-disable-checker ericsson.misrac.SwitchDefaultBranch -analyzer-disable-checker ericsson.mtas.DbnDelayAfterRetry -analyzer-disable-checker ericsson.mtas.DialogueSetupCreate -analyzer-disable-checker ericsson.mtas.EpctParameters -analyzer-disable-checker ericsson.mtas.IllegalTracing -analyzer-disable-checker ericsson.mtas.LicenseManagement -analyzer-disable-checker ericsson.mtas.MtasConventions -analyzer-disable-checker ericsson.mtas.SerializeVersion -analyzer-disable-checker ericsson.mtas.SerializeWithoutObserver -analyzer-checker=ericsson.precpp11.stl.AllocWithState -analyzer-checker=ericsson.precpp11.stl.BinFunctorTypeMismatch -analyzer-checker=ericsson.precpp11.stl.UnaryFunctorTypeMismatch -analyzer-disable-checker ericsson.tsp.DbnTransactionManagement -analyzer-disable-checker ericsson.tsp.PointerCompare -analyzer-disable-checker ericsson.tsp.TspBuiltinTypes -analyzer-disable-checker ericsson.utility.WarningToReportConverter -analyzer-disable-checker llvm.Conventions -analyzer-checker=nullability.NullPassedToNonnull -analyzer-checker=nullability.NullReturnedFromNonnull -analyzer-disable-checker nullability.NullableDereferenced -analyzer-disable-checker nullability.NullablePassedToNonnull -analyzer-disable-checker nullability.NullableReturnedFromNonnull -analyzer-disable-checker optin.cplusplus.VirtualCall -analyzer-disable-checker optin.mpi.MPI-Checker -analyzer-disable-checker optin.osx.cocoa.localizability.EmptyLocalizationContextChecker -analyzer-disable-checker optin.osx.cocoa.localizability.NonLocalizedStringChecker -analyzer-disable-checker optin.performance.Padding -analyzer-disable-checker optin.portability.UnixAPI -analyzer-disable-checker osx.API -analyzer-disable-checker osx.NumberObjectConversion -analyzer-disable-checker osx.ObjCProperty -analyzer-disable-checker osx.SecKeychainAPI -analyzer-disable-checker osx.cocoa.AtSync -analyzer-disable-checker osx.cocoa.ClassRelease -analyzer-disable-checker osx.cocoa.Dealloc -analyzer-disable-checker osx.cocoa.IncompatibleMethodTypes -analyzer-disable-checker osx.cocoa.Loops -analyzer-disable-checker osx.cocoa.MissingSuperCall -analyzer-disable-checker osx.cocoa.NSAutoreleasePool -analyzer-disable-checker osx.cocoa.NSError -analyzer-disable-checker osx.cocoa.NilArg -analyzer-disable-checker osx.cocoa.NonNilReturnValue -analyzer-disable-checker osx.cocoa.ObjCGenerics -analyzer-disable-checker osx.cocoa.RetainCount -analyzer-disable-checker osx.cocoa.SelfInit -analyzer-disable-checker osx.cocoa.SuperDealloc -analyzer-disable-checker osx.cocoa.UnusedIvars -analyzer-disable-checker osx.cocoa.VariadicMethodTypes -analyzer-disable-checker osx.coreFoundation.CFError -analyzer-disable-checker osx.coreFoundation.CFNumber -analyzer-disable-checker osx.coreFoundation.CFRetainRelease -analyzer-disable-checker osx.coreFoundation.containers.OutOfBounds -analyzer-disable-checker osx.coreFoundation.containers.PointerSizedValues -analyzer-checker=security.FloatLoopCounter -analyzer-checker=security.insecureAPI.UncheckedReturn -analyzer-checker=security.insecureAPI.getpw -analyzer-checker=security.insecureAPI.gets -analyzer-checker=security.insecureAPI.mkstemp -analyzer-checker=security.insecureAPI.mktemp -analyzer-checker=security.insecureAPI.rand -analyzer-disable-checker security.insecureAPI.strcpy -analyzer-checker=security.insecureAPI.vfork -analyzer-checker=unix.API -analyzer-checker=unix.Malloc -analyzer-checker=unix.MallocSizeof -analyzer-checker=unix.MismatchedDeallocator -analyzer-checker=unix.StdCLibraryFunctions -analyzer-checker=unix.Vfork -analyzer-checker=unix.cstring.BadSizeArg -analyzer-checker=unix.cstring.NullArg -analyzer-checker=valist.CopyToSelf -analyzer-checker=valist.Uninitialized -analyzer-checker=valist.Unterminated -analyzer-config xtu-dir=/local/scratch/repos/ML66/reports/bns__2017_11_16__15_34_00/ctu-dir/i386 -analyzer-display-ctu-progress -o /local/scratch/repos/ML66/reports/bns__2017_11_16__15_34_00/ArpInterface.cc_d8f953b6e736cd3e4069dfea22a426b1.plist -x c++ /local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/platform/lch/lh/src/ArpInterface.cc 
1.	<eof> parser at end of file
2.	While analyzing stack: 
	#0 void Interface::clearArpCache()
3.	/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/platform/lch/lh/src/ArpInterface.cc:59:5: Error evaluating statement
4.	/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/platform/lch/lh/src/ArpInterface.cc:59:5: Error evaluating statement
clang-5.0: error: unable to execute command: Segmentation fault
clang-5.0: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 5.0.0 
Target: powerpc-montavista-linux-gnu
Thread model: posix
InstalledDir: /proj/tn/tools/code_checker/CodeChecker_6.2.0/cc_bin
clang-5.0: note: diagnostic msg: PLEASE submit a bug report to http://llvm.org/bugs/ and include the crash backtrace, preprocessed source, and associated run script.
clang-5.0: error: unable to execute command: Segmentation fault
clang-5.0: note: diagnostic msg: Error generating preprocessed source(s).
```

--- a/lib/AST/ASTImporter.cpp
+++ b/lib/AST/ASTImporter.cpp
@@ -78,7 +78,8 @@ namespace clang {
     QualType VisitElaboratedType(const ElaboratedType *T);
     QualType VisitDependentNameType(const DependentNameType *T);
     QualType VisitPackExpansionType(const PackExpansionType *T);
-    // FIXME: DependentTemplateSpecializationType
+    QualType VisitDependentTemplateSpecializationType(
+        const DependentTemplateSpecializationType *T);
     QualType VisitObjCInterfaceType(const ObjCInterfaceType *T);
     QualType VisitObjCObjectType(const ObjCObjectType *T);
     QualType VisitObjCObjectPointerType(const ObjCObjectPointerType *T);
@@ -854,6 +855,25 @@ QualType ASTNodeImporter::VisitPackExpansionType(const PackExpansionType *T) {
 
   return Importer.getToContext().getPackExpansionType(Pattern,
                                                       T->getNumExpansions());
+}
+
+QualType ASTNodeImporter::VisitDependentTemplateSpecializationType(
+    const DependentTemplateSpecializationType *T) {
+  NestedNameSpecifier *Qualifier = Importer.Import(T->getQualifier());
+  if (!Qualifier && T->getQualifier())
+    return QualType();
+
+  IdentifierInfo *Name = Importer.Import(T->getIdentifier());
+  if (!Name && T->getIdentifier())
+    return QualType();
+
+  SmallVector<TemplateArgument, 2> ToPack;
+  ToPack.reserve(T->getNumArgs());
+  if (ImportTemplateArguments(T->getArgs(), T->getNumArgs(), ToPack))
+    return QualType();
+
+  return Importer.getToContext().getDependentTemplateSpecializationType(
+        T->getKeyword(), Qualifier, Name, ToPack);
 }
 
 QualType ASTNodeImporter::VisitObjCInterfaceType(const ObjCInterfaceType *T) {

--- a/unittests/AST/ASTImporterTest.cpp
+++ b/unittests/AST/ASTImporterTest.cpp
@@ -528,6 +528,24 @@ TEST(ImportType, ImportPackExpansion) {
                                          declRefExpr()))))))))));
 }
 
+const internal::VariadicDynCastAllOfMatcher<Type,
+                                            DependentTemplateSpecializationType>
+    dependentTemplateSpecializationType;
+
+TEST(ImportType, ImportDependentTemplateSpecialization) {
+  MatchVerifier<Decl> Verifier;
+  EXPECT_TRUE(testImport(
+      "template<typename T>"
+      "struct A;"
+      "template<typename T>"
+      "struct declToImport {"
+      "  typename A<T>::template B<T> a;"
+      "};",
+      Lang_CXX, "", Lang_CXX, Verifier,
+      classTemplateDecl(has(cxxRecordDecl(
+          has(fieldDecl(hasType(dependentTemplateSpecializationType()))))))));
+}
+
 const internal::VariadicDynCastAllOfMatcher<Stmt, SizeOfPackExpr>
     sizeOfPackExpr;
 

